src/user-guide/running-workflows/reflow.rst

```diff
@@ -1,86 +1,227 @@
-   because they are considered to be an unfinished part of the existing flow.
 
-Reflow Use Cases
-----------------
+Flow Numbers
+------------
 
-Reflows are useful if you need to re-wind your :term:`workflow` to allow
-it to evolve a new path into the future, or to repeat-run some subgraph
-to regenerate its outputs after making changes.
+Flows are identified by numbers passed down from parent task to child task in
+the graph.
 
-For example, the following :term:`cycling workflow` runs a :term:`task`
-called ``model`` in every cycle, followed by a postprocessing task, two
-product generating tasks, and finally a task that publishes results for
-the cycle point:
+Flow number ``1`` is triggered automatically by ``cylc play`` at :term:`scheduler`
+start-up. The next flow started by :ref:`manual triggering<triggering-flows>`
+gets the number ``2``, then ``3``, and so on.
 
-.. code-block:: cylc
+Tasks can carry multiple flow numbers as a result of
+:ref:`flow merging<flow-merging>` and :ref:`manual triggering<triggering-flows>`.
 
-   [scheduling]
-       cycling mode = integer
-       initial cycle point = 1
-       [[graph]]
-           P1 = model[-P1] => model => post => prod1 & prod2 => publish
+.. note::
+   Flow numbers are not yet exposed in the UI, but they are logged with task
+   events in the :term:`scheduler log`.
 
-Let's say the workflow has run to cycle 8, but we have just noticed that
-a corrupted ancillary file resulted in bad products at cycle 5.
 
-To rectify this we could fix the corrupted file and trigger a new flow
-(a reflow) from ``5/post``::
+.. _triggering-flows:
 
-   cylc trigger --reflow <workflow-id>//5/post
+Triggering & Flows
+------------------
 
-The new flow will regenerate and republish cycle 5 products before naturally
-coming to a halt, because the triggered tasks do not lead on to the next cycle
-and the rest of the graph.
+By default, manual triggering (with ``cylc trigger`` or the UI) starts a new
+:term:`front of activity<flow front>` in current flows.
+But it can also start new flows and trigger flow-independent single tasks.
 
-Meanwhile, the original flow will carry on unaffected, from cycle 8.
+In the diagrams below, the grey nodes represent tasks in flow ``1``, and
+the blue ones stem from a single manual triggering event.
 
-Flow Merging
-------------
+Triggering in Current Flows
+   ``cylc trigger [--wait] ID``
+
+   This is the default trigger action: trigger the task and give it all current
+   active flow numbers. Subsequently, each of those flows will consider this
+   task to have run already.
+
+   **Ahead of active flows** this starts a new front of activity for the active
+   flows, which by default can continue on without waiting for catch up.
+
+   .. image:: ../../img/same-flow-n.png
+
+   With ``--wait``, the result is the same except that any action downstream of 
+   the triggered task is delayed until the first flow catches it.
```

Comment:
I think this needs a separate diagram - only the one task should be shown in blue?
I'm also not sure a single diagram really illustrates what's going on.
Could we use animates gif's?
Is there any way we could generate this diagrams?
Follow-up issue?